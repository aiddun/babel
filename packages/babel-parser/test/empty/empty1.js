import { parse } from "../../lib";

function getParser(code) {
  return () => parse(code, { sourceType: "module" });
}

describe("empty symbol assignment", function () {
  it("should parse", function () {
    expect(getParser(`const a = [...]`)()).not.toThrow();
  });
});

describe("empty symbol with other elements", function () {
  it("should parse", function () {
    expect(getParser(`const a = [a, b, ..., d]`)()).not.toThrow();
  });
});

describe("empty symbol ternary", function () {
  it("should parse", function () {
    expect(getParser(`[a ? b : ...]`)()).not.toThrow();
  });
});
