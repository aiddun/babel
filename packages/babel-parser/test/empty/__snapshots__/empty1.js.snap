// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`empty symbol should parse 1`] = `
Node {
  "comments": Array [],
  "end": 5,
  "errors": Array [],
  "loc": SourceLocation {
    "end": Position {
      "column": 5,
      "line": 1,
    },
    "filename": undefined,
    "identifierName": undefined,
    "start": Position {
      "column": 0,
      "line": 1,
    },
  },
  "program": Node {
    "body": Array [
      Node {
        "end": 5,
        "expression": Node {
          "elements": Array [
            Node {
              "end": 4,
              "loc": SourceLocation {
                "end": Position {
                  "column": 4,
                  "line": 1,
                },
                "filename": undefined,
                "identifierName": undefined,
                "start": Position {
                  "column": 1,
                  "line": 1,
                },
              },
              "start": 1,
              "type": "SpreadElement",
            },
          ],
          "end": 5,
          "loc": SourceLocation {
            "end": Position {
              "column": 5,
              "line": 1,
            },
            "filename": undefined,
            "identifierName": undefined,
            "start": Position {
              "column": 0,
              "line": 1,
            },
          },
          "start": 0,
          "type": "ArrayExpression",
        },
        "loc": SourceLocation {
          "end": Position {
            "column": 5,
            "line": 1,
          },
          "filename": undefined,
          "identifierName": undefined,
          "start": Position {
            "column": 0,
            "line": 1,
          },
        },
        "start": 0,
        "type": "ExpressionStatement",
      },
    ],
    "directives": Array [],
    "end": 5,
    "interpreter": null,
    "loc": SourceLocation {
      "end": Position {
        "column": 5,
        "line": 1,
      },
      "filename": undefined,
      "identifierName": undefined,
      "start": Position {
        "column": 0,
        "line": 1,
      },
    },
    "sourceType": "module",
    "start": 0,
    "type": "Program",
  },
  "start": 0,
  "type": "File",
}
`;
